### PR TITLE
Namespaced NSQ

### PR DESCRIPTION
**Pull request by nickstenning**
To solve the issue of different instances of `h` stomping all over one another when they share an NSQ cluster, add support for setting a "namespace" which will be prefixed to all the topic names that queue readers and writers address.

### CODE REVIEW

**Comment by nickstenning** [comment]
Oh, and @tilgovi, I did see your note about adding appropriate metadata to queue messages to prevent this problem at a higher level. I'm not so sure about that. It means, for example, that production instances would have to actually read and deserialize every message from staging: all so they can inspect the "domain" field and then throw the message away. Seems like unnecessary network and CPU overhead that can be avoided by this change and setting `NSQ_NAMESPACE=stage` on the stage application.

**Comment by gergely-ujvari** [comment]
LGTM (but I know @tilgovi wants to review this too)

File: h/queue.py [hunk 1]
```diff
@@ -11,9 +22,11 @@ def get_reader(request, topic, channel):
     The caller is responsible for adding appropriate `on_message` hooks and
     starting the reader.
     """
-    setting = request.registry.settings.get('nsq.reader.addresses',
-                                            'localhost:4150')
-    addrs = aslist(setting)
+    ns = request.registry.settings.get('nsq.namespace')
```
Comment (tilgovi) [inline h/queue.py]: can you provide a default? Otherwise we get `-topic` at line 29.

File: h/queue.py [hunk 2]
```diff
@@ -24,10 +37,11 @@ def get_writer(request):
     writer address configured in settings. The writer communicates over the
     nsq HTTP API and does not hold a connection open to the nsq instance.
     """
-    setting = request.registry.settings.get('nsq.writer.address',
-                                            'localhost:4151')
-    hostname, port = setting.split(':', 1)
-    nsqd = gnsq.Nsqd(hostname, http_port=port)
+    ns = request.registry.settings.get('nsq.namespace')
```
Comment (tilgovi) [inline h/queue.py]: The same here.

**Comment by tilgovi** [comment]
Otherwise, LGTM

File: h/queue.py (hunk 1 above)
Comment (nickstenning) [inline h/queue.py]: No you don't. `.get()` returns `None` by default.

File: h/queue.py (hunk 2 above)
Comment (nickstenning) [inline h/queue.py]: Ditto.

**Pull request merged**